goosebit/api/download.py

```diff
@@ -1,20 +1,29 @@
-from fastapi import APIRouter, Security
+from fastapi import APIRouter, HTTPException
 from fastapi.requests import Request
-from fastapi.responses import FileResponse
+from fastapi.responses import FileResponse, RedirectResponse
 
-from goosebit.auth import validate_user_permissions
-from goosebit.permissions import Permissions
+from goosebit.models import FirmwareUpdate
 from goosebit.settings import UPDATES_DIR
 
 router = APIRouter(prefix="/download")
 
 
-@router.get(
-    "/{file}",
-    dependencies=[
-        Security(validate_user_permissions, scopes=[Permissions.FIRMWARE.READ])
-    ],
-)
-async def download_file(request: Request, file: str):
-    filename = UPDATES_DIR.joinpath(file)
-    return FileResponse(filename, media_type="application/octet-stream")
+@router.get("/by_name/{filename}")
```

Comment:
What will this endpoint be used for?

Reply:
Fair enough, I think this can safely be removed, as all firmware can now just be queried by ID, even in the `/firmware/all` endpoint it reports the ID.